internal/testutils/endpoint.go

```diff
@@ -17,14 +17,39 @@ limitations under the License.
 package testutils
 
 import "github.com/kubernetes-incubator/external-dns/endpoint"
+import "sort"
 
 /** test utility functions for endpoints verifications */
 
+type byAllFields []*endpoint.Endpoint
+
+func (b byAllFields) Len() int      { return len(b) }
+func (b byAllFields) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byAllFields) Less(i, j int) bool {
+	if b[i].DNSName < b[j].DNSName {
+		return true
+	}
+	if b[i].DNSName == b[j].DNSName {
+		if b[i].Target < b[j].Target {
+			return true
+		}
+		if b[i].Target == b[j].Target {
+			if b[i].RecordType <= b[j].RecordType {
```

Comment:
return b[i].RecordType <= b[j].RecordType

and drop the other 2 next return lines